src/riak_kv_delete.erl

```diff
@@ -60,7 +60,10 @@ delete(ReqId,Bucket,Key,Options,Timeout,Client,ClientId,undefined) ->
         {R, PR, PassThruOpts} ->
             RealStartTime = riak_core_util:moment(),
             {ok, C} = riak:local_client(),
-            case C:get(Bucket,Key,[{r,R},{pr,PR},{timeout,Timeout}]++PassThruOpts) of
+            %% a TS bucket is a {Table, Table}
+            DupBucketForTS = if is_binary(Key) -> Bucket; el/=se -> {Bucket, Bucket} end,
```

Comment:
Instead of having a TS-specific clause in the `riak_kv_delete` code, I'm thinking we could do a better job encapsulating the TS code if we converted the table to a bucket in `riak_kv_pb_timeseries`. That way the existing `riak_kv_delete:delete` function doesn't need to be modified with a special case.